### DISCUSSION TITLE
How to use queryContent ?

### DISCUSSION THREAD

**Discussion by hlassiege** [nuxt/nuxt#15937] [Questions] [answered]
Hi,

I have a personal blog that runs with nuxt 2 and nuxt content.
I am currently trying to migrate to nuxt 3 and nuxt content 2 but I'm not sure how to use queryContent and asyncData.

In the old code, it was written this way : 

```
  async asyncData({$content}) {
    const article = await $content("resources")
      .where({ title: 'Ressources' })
      .fetch();
    return {
      article: article[0],
    };
  },
```

The content folder contains : 

```
content 
|
`------ articles  (not relevant for the moment)
`------ resources
     `------ resources.md
```

the document resources.md has the following yaml header : 

```
---
title: "Ressources"
img: ""
cover: "cover1.jpg"
---
```

In nuxt 3, I had to use composition API (I don't know if it was mandatory but I didn't succeed to adapt the old code. 
I'm not familiar with it but I assumed I can have two scripts in the same file 
(the full file is here : https://github.com/hlassiege/eventuallycoding/blob/nuxt3/pages/resources.vue) 

So I transformed the code into : 

```
const { article } = await useAsyncData('Resources', () => queryContent('resources')
    // .where({ title: 'Ressources' })
    .findOne());
console.log(article);
```

However, article is always undefined after that. So I guess I don't understand how to use queryContent. 
Do you know what could be wrong with this code above ?

**Comment by JackEdwardLyons** [comment]
I'm facing the same issue. Any news or helpful tips on this would be much appreciated!

**Comment by motea927** [comment]
Hi @hlassiege, 
After I clone your project, 
I find that `queryContent` is not working,
I think you have some issue version dependencies,
So after removing `package-lock.json` and `node_modules`,
And re install dependencies,
It will work.

**Comment by hlassiege** [comment, marked as answer]
yes sorry, the issue was not updated. Indeed, everything works. The blog is running on nuxt3 here: https://eventuallycoding.com/
I even blogged about the migration itself ^^ (https://eventuallycoding.com/2022/11/blog-migration-on-nuxt, in french sorry)
